src/validators/EmailValidator.js

```diff
@@ -0,0 +1,9 @@
+const validEmailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```

Comment:
Did you write this or find it? If the later, let's leave a link back to it in a comment.

just looking at `^[<>()[\]\\.,;:\s@"]+` takes a lot of effort to mentally parse the character class and how it'd map to the first part of the email before `@` (I think). With complex regex I think documentation and links back to resources is really useful.

Reply:
Added a link to where I got it from - tested a few different versions and this one seemed most comprehensive.